Vizzuality/heco-invest
# Dashboard layout / page placeholders w/permissions set

## Description

**In this PR:**  

- Added back a `Naked` layout   
  _In many cases we need to pass dynamic props to a layout. This can't be done the way we have the frontend set up, so it is meant to be used as an "Empty" layout of sorts so we can use a Layout as a component._    

  _Form pages could be switched to this layout in the future as well, as it is identical. Switch not made in this PR because it's a huge PR already, and to keep things separated._

- Added the `Dashboard` layout  
  - Has a `buttons` prop so we specify page-specific buttons to be displayed in the header
    _Eg: `New project`_  
  - Has basic responsiveness implemented  
    _On small screen devices, only the top part will stick to the top to mimic the behaviour of the static pages_  
    _On larger screen devices, the whole header including dashboard specific navigation should stick to the top_  
  - Uses the `ProtectedPage` layout as a wrapper. It is set up to guard against:  
    - Non signed in users
      _will redirect users to sign-in page_  
    - Accounts that are not project developers nor investors  
      _Guarding against not yet approved accounts to be done separately, as such a guard has not yet been implemented and is needed throughout the app._  
  - Includes the following separate components:  
    - `Header`  
      _Very similar to the static pages header._
    - `AccountInfo`
      _With the account name and a link to the user public page (investor or project developer one)_  
    - `AccountPicture`  
      _Responsible for displaying the user's account picture, with a placeholder as a fallback in case the user doesn't have one set up or there is an error loading their picture_  
    - `Navigation`  
      _Shared navigation component for both project developer and investor dashboard views. Which are displayed depends on the current user's role, because project developers and investors will have access to different dashboard pages_
      _Accessible and friendly to screen readers_  
- Placeholder `dashboard pages`  
  _For both project developer and investor accounts, already set up with the `ProtectedPage` component to restrict access of each user role to only the pages they should have access to._  

**Others:**  

- New `useScrollY` hook  
  _In certain layouts we use the `useWindowScrollPosition` from `rooks` in order to stick/change the background of the header navigation. Since we have to check for whether the `window` object is defined or not, I've added an easy to use hook to make these types of checks cleaner and de-duplicate code._  
- Fixed an issue causing occasional crashes in the `ProjectCard` component  
  _When we fail to fetch enums from the API_
  _Not directly related to this PR, but it kept happening frequently while developing the dashboard layout_

## Testing instructions

Verify that:  
   - When not signed in, all the dashboard pages are inaccessible and you will be redirected to the sign in page  
   - When signed in as a project developer, you only have access to the project developer specific dashboard pages 
   - When signed in as a project developer, only the project developer specific links are shown  
   - When signed in as a investor, you only have access to the investor specific dashboard pages 
   - When signed in as a investor, only the investor specific links are shown  
   - The header is useable on mobile   

## Tracking

[LET-549](https://vizzuality.atlassian.net/browse/LET-549)

## Screenshot

<img width="1680" alt="Screenshot 2022-06-02 at 15 36 18" src="https://user-images.githubusercontent.com/6273795/171654248-42171932-9083-4d38-b115-6fa1ab5c2641.png">
